tests/spaces/test_spaces.py

```diff
@@ -448,3 +448,49 @@ def test_space_legacy_state_pickling():
     assert space._np_random == legacy_state["np_random"]
     assert space.n == 3
     assert space.dtype == legacy_state["dtype"]
+
+
+@pytest.mark.parametrize(
+    "space",
+    [
+        Box(low=0, high=np.inf, shape=(2,), dtype=np.int64),
+        Box(low=0, high=np.inf, shape=(2,), dtype=np.float64),
+        Box(low=-np.inf, high=0, shape=(2,), dtype=np.int64),
+        Box(low=-np.inf, high=0, shape=(2,), dtype=np.float64),
+        Box(low=0, high=np.inf, shape=(2,), dtype=np.int32),
+        Box(low=0, high=np.inf, shape=(2,), dtype=np.float32),
+        Box(low=-np.inf, high=0, shape=(2,), dtype=np.int32),
+        Box(low=-np.inf, high=0, shape=(2,), dtype=np.float32),
+    ],
+)
+def test_infinite_space(space):
+    # for this test, make sure that spaces that are passed in have only 0 or infinite bounds
+    # because space.high and space.low are both modified within the init
+    # so the only way to test that infinite bounded space are truly infinite during test
+    # is to ensure that we only pass in either infinite or 0
+    assert np.all(space.high > space.low), "High bound not higher than low bound"
+    assert space.contains(space.sample())
```

Comment:
Add `space.seed(0)` before this, at the moment this is technically a stochastic test, so it sometimes might work and sometimes fail. Also I'd like to see some more assertions on what this is expected to be, e.g. that the dtype is right, and that the value is right (e.g. for the first two spaces tested, it should be positive - in principle `contains` should test it, but we're messing with the bounds, so a simple sanity check would be valuable.

It'd be good to also vary the shapes a bit, I'd like to see at least one test where it's a box larger than 1D (so e.g. `shape=(2,3)`)
From what I can see in the code, it's handled properly so it should still pass, but I can easily see a naive implementation of this that could pass your current tests, but fail on the example I gave, so it's good to cover our bases